tools/cell_census_builder/validate.py

```diff
@@ -447,6 +445,45 @@ def validate_manifest_contents(assets_path: str, datasets: List[Dataset]) -> boo
     return True
 
 
+def validate_consolidation(soma_path: str, experiment_builders: List[ExperimentBuilder]) -> bool:
+    """Verify that obs, var and X layers are all fully consolidated & vacuumed"""
+    with soma.Collection.open(soma_path, context=SOMA_TileDB_Context()) as census:
+        consolidated_uris = list_uris_to_consolidate(census)
+        for uri in consolidated_uris:
+            assert len(tiledb.array_fragments(uri)) == 1, f"{uri} has not been fully consolidated & vacuumed"
+    return True
+
+
+def validate_directory_structure(soma_path: str, assets_path: str) -> bool:
+    """Verify that the entire census is a single directory tree"""
+    assert soma_path.startswith(assets_path.rsplit("/", maxsplit=1)[0])
+    assert os.path.exists(soma_path) and os.path.exists(assets_path)
+    assert soma_path.endswith("soma") and assets_path.endswith("h5ads")
+    return True
+
+
+def validate_relative_path(soma_path: str) -> bool:
+    """
+    Verify the census objects are stored in the same relative path
+    :param soma_path:
+    :return:
+    """
+    # TODO use SOMA API. See https://github.com/single-cell-data/TileDB-SOMA/issues/999
+
+    def _walk_tree(name: str, parent: Any) -> None:
+        if parent.soma_type in ["SOMACollection", "SOMAExperiment", "SOMAMeasurement"]:
```

Comment:
There is absolutely _nothing_ wrong with this code, but I have small suggestion to make it a bit more elegant/concise.  Completely optional!

You can do the same by using the Python type system and more concisely, written as:

> if isinstance(parent, soma.Collection):

This is because Experiment & Measurement are Collection sub-classes in the Python typing.